README.md

```diff
@@ -0,0 +1,73 @@
+# Overview
+
+This is the base layer for charms that wish to take advantage of
+Apache's Bigtop framework for configuring and deploying services via
+puppet. [Including this layer][building] gives you access to a Python
+class called Bigtop, which contains various useful tools, including an
+automatically run install routine.
+
+# Usage
+
+To create a charm using this base layer, first you must include it in
+your `layer.yaml` file:
+
+```yaml
+includes: ['layer:apache-bigtop-base']
+```
+
+This will fetch the layer from [interfaces.juju.solutions][] and
+incorporate it into your charm. To use the Bigtop class, import it,
+then call its .render_site_yaml and .trigger_puppet routines in
+sequence, like so:
+
+```python
+from charms.layer.apache_bigtop_base import Bigtop
+
+# Setup arguments to pass to .render_site_yaml
+hosts = {'myservice': <host>}
+roles = ['myservice-worker', 'myservice-client']
+override = {'myservice::somevalue': <somevalue>, ...}
+
+# Trigger a puppet run
+bigtop = Bigtop()
+bigtop.render_site_yaml(hosts, roles, override)
+bigtop.trigger_puppet()
+```
+
+This tells Bigtop to run puppet and install your service.
+
+Bigtop can either infer which services you wish to install by
+inspecting the "hosts" that you pass to .render_site_yaml, or it can
+install the specific services that you name in "roles".
+
+Of course, Bigtop needs to know how to deploy your service!
```

Comment:
There are two service deployment methods for bigtop. You can either specify the roles to have a more fine grained control on what is deployed, or you can specify a component for with ultimately an predefined set of roles is configured. Our component deployment path is not well tested, so it might not be functional at the moment.